### ISSUE TITLE
Lossless compression for textures

### ISSUE THREAD

**Issue by H-A-M-G-E-R** [MonoGame/MonoGame#8470]
### Intent

DXT is a lossy compression algorithm, but I want the pipeline to process to a losslessly compressed texture, like QOA, which decompresses faster than PNG.

### Motivation

i hate lossily compressed images in 2D games

**Comment by H-A-M-G-E-R** [comment]
It already exists: https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework.Content.Pipeline/Graphics/TextureProfile.cs#L52
but clarify that DXT has lossy compression

**Comment by mrhelmut** [comment]
While it is relevant that the documentation should specify that texture compression is always lossy, it seems that there is a misunderstanding between file compression and GPU compression.

```Texture compression``` always refers to how texture are stored inside the GPU memory, not how it is stored on drive as a file (this is file compression; although texture compression exist on drive as well, it is not their main purpose).

PNG and QOI (I guess you didn't mean QOA which is the equivalent for audio) are _file_ compression formats. Those are not directly readable by a GPU.

DXT is a _GPU_ compression format. GPU can read it almost for free. The tradeoff being that all GPU compression formats are lossy (if we exclude runtime decoders).

SpriteFont defaults to compressed because they tend to be quite large.

----

Leaving this open to enhance the documentation.

**Comment by H-A-M-G-E-R** [comment]
And if you want a smaller size than SpriteFonts use an external library to load TTF or OTF natively. I highly against SpriteFonts.

**Comment by mrhelmut** [comment]
> Does it mean that the PNGs are loaded and compressed to DXT in memory when loading the game?

When you provide a PNG to the content pipeline to turn it into an XNB, the PNG data will no longer exist and are converted to the selected texture format.
If you select DXT, the image data will be DXT both on drive inside the XNB, and the GPU memory.
If you select COLOR, the image data will be stored pretty much like it'd be BMP data both on drive, and the GPU memory.

The conversion doesn't happen when the game loads the XNB. It happens when the XNB is created and the game doesn't convert anything, it uses the data as-is from the XNB directly into the GPU.

> And if you want a smaller size than SpriteFonts use an external library to load TTF or OTF natively. I highly against SpriteFonts.

The external library will do the exact same thing, although hidden from you but in the end a spritesheet will be created into GPU memory and filled with rasterized characters (and the library will likely create a separate spritesheet for each requested sizes; unless it uses something like MSDF but I don't think there are libraries around for MonoGame that does on a runtime basis).

MonoGame can indeed do better in that regard.

**Comment by stromkos** [comment]
> Does it mean that the PNGs are loaded and compressed to DXT in memory when loading the game?

No, no change to any format other than `Color[]` is **ever** done by the framework code at run-time.

The default format is a Color Array, no loss for <16K images.

Unless you select one of the DXT formats during the texture build process it will be a Color array. 

---

>but I want the pipeline to process to a losslessly compressed texture, like QOI

Sure,  You are welcome to add the "QOI" format for processing to a `Color[]` into an .XNB file, and speed up the build process, but this will always be the same as a BMP for loading times.

---

>     I highly against SpriteFonts.

Sure, store all Text in images for a single language experience, instead of building a single `SpriteFont`. It is much easier to support a single language  to  "against SpriteFonts"

I use platform specific replacement for Emojis, so I understand the inclination, each project is unique.

**Comment by H-A-M-G-E-R** [comment]
> Sure, You are welcome to add the "QOI" format for processing to a Color[] into an .XNB file, and speed up the build process, but this will always be the same as a BMP for loading times.

FNA already does this, but for raw files without the XNB wrapper.
